usr/share/rear/prep/default/310_include_uefi_tools.sh

```diff
@@ -5,10 +5,13 @@ if grep -qw 'noefi' /proc/cmdline; then
     return
 fi
 
-# next step, is checking /boot/efi directory case-insensitive for the /EFI part (we need it)
-if [[ $(find /boot -maxdepth 1 -iname efi -type d | wc -l) -eq 0 ]] ; then
-    return    # must be mounted
-fi
+# Next step, is checking /boot/efi directory case-insensitive for the /EFI part (we need it).
+# If no /boot/[eE][fF][iI] directory can be found we cannot copy the UEFI binaries we might need.
```

Comment:
Why not simply `[[ -d /boot/[eE][fF][iI] ]]` ?? In any case all this code would fail if there is more than one match (e.g. efi and EFI).

IMHO much more problematic is the hard-coded assumption that the EFI Boot partition is mounted on `/boot/efi`. While this might hold true for regular distros, I wound't be so sure about Arch and Gentoo. With [Gummiboot/systemd-boot](https://www.freedesktop.org/wiki/Software/systemd/systemd-boot/) the layout will be for sure different as all boot files must reside on the ESP.

Also, a quick check revealed that we have hard coded `/boot/efi` in many other places:
- Search for [boot efi](https://github.com/rear/rear/search?utf8=%E2%9C%93&q=boot+efi&type=Code)
- Search for [oot efi](https://github.com/rear/rear/search?utf8=%E2%9C%93&q=oot+efi&type=Code)

I guess the only "proper" solution is to follow the same logic that the UEFI BIOS uses:

1. Identify the [EFI System Partition](https://en.wikipedia.org/wiki/EFI_system_partition)
2. Determine its mount point
3. Use that instead of `/boot/efi`